macros/parsers/parserGraphTool.pl

```diff
@@ -798,124 +797,129 @@ sub ans_rule {
+			$tikz .=
+				"\\foreach \\y in {"
+				. join(',', @yTicks)
+				. "}{\\draw[thin] (5pt,\\y) -- (-5pt,\\y) node[left]{\$\\y\$};}\n"
+				if (@yTicks);
+		}
 
-				# Switch from the foreground layer to the background layer for the fills.
-				$tikz .= "\\end{pgfonlayer}\n\\begin{pgfonlayer}{background}\n";
+		# Border box
+		$tikz .= "\\draw[borderblue,rounded corners=14pt,thick] "
+			. "($self->{bBox}[0],$self->{bBox}[3]) rectangle ($self->{bBox}[2],$self->{bBox}[1]);\n";
 
-				# Now shade the fill regions.
-				for (@{ $obj->{data} }) {
-					next
-						unless (ref($graphObjectTikz{ $_->{data}[0] }) eq 'HASH'
-							&& ref($graphObjectTikz{ $_->{data}[0] }{code}) eq 'CODE'
-							&& $graphObjectTikz{ $_->{data}[0] }{fillType});
-					$tikz .= $graphObjectTikz{fill}{code}->($self, $_, [@obj_data], $obj);
-				}
+		# Graph the points, lines, circles, and parabolas.
+		if (@{ $self->{staticObjects} }) {
+			my $obj = $self->SUPER::new($self->{context}, @{ $self->{staticObjects} });
+
+			# Switch to the foreground layer and clipping box for the objects.
+			$tikz .= "\\begin{pgfonlayer}{foreground}\n";
+			$tikz .= "\\clip[rounded corners=14pt] "
+				. "($self->{bBox}[0],$self->{bBox}[3]) rectangle ($self->{bBox}[2],$self->{bBox}[1]);\n";
 
-				# End the background layer.
-				$tikz .= "\\end{pgfonlayer}";
+			my @obj_data;
+
+			# First graph lines, parabolas, and circles.  Cache the clipping path and a function
+			# for determining which side of the object to shade for filling later.
+			for (@{ $obj->{data} }) {
+				next
+					unless (ref($graphObjectTikz{ $_->{data}[0] }) eq 'HASH'
+						&& ref($graphObjectTikz{ $_->{data}[0] }{code}) eq 'CODE'
+						&& !$graphObjectTikz{ $_->{data}[0] }{fillType});
+				my ($object_tikz, $object_data) = $graphObjectTikz{ $_->{data}[0] }{code}->($self, $_);
+				$tikz .= $object_tikz;
+				push(@obj_data, $object_data);
 			}
 
-			$graph->tex($tikz);
+			# Switch from the foreground layer to the background layer for the fills.
+			$tikz .= "\\end{pgfonlayer}\n\\begin{pgfonlayer}{background}\n";
 
-			$out = main::image(
-				main::insertGraph($graph),
-				width    => $size[0],
-				height   => $size[1],
-				tex_size => $self->{texSize}
-			);
+			# Now shade the fill regions.
+			for (@{ $obj->{data} }) {
+				next
+					unless (ref($graphObjectTikz{ $_->{data}[0] }) eq 'HASH'
+						&& ref($graphObjectTikz{ $_->{data}[0] }{code}) eq 'CODE'
+						&& $graphObjectTikz{ $_->{data}[0] }{fillType});
+				$tikz .= $graphObjectTikz{fill}{code}->($self, $_, [@obj_data], $obj);
+			}
+
+			# End the background layer.
+			$tikz .= "\\end{pgfonlayer}";
 		}
+
+		$graph->tex($tikz);
+
+		my $out = main::image(
+			main::insertGraph($graph),
+			width    => $size[0],
+			height   => $size[1],
+			tex_size => $self->{texSize}
+		);
+		$out .= qq!<p><fillin name="$ans_name"/></p>! if ($main::displayMode eq 'PTX');
+		return $out;
```

Comment:
This could be
```suggestion
		return main::image(
			main::insertGraph($graph),
			width    => $size[0],
			height   => $size[1],
			tex_size => $self->{texSize}
		) . ($main::displayMode eq 'PTX' ? qq!<p><fillin name="$ans_name"/></p>! : '');
```